### DISCUSSION TITLE
OML 2.0 Scope

### DISCUSSION THREAD

**Discussion by melaasar** [opencaesar/platform#2] [Announcements]
## Summary
The openCAEASAR team is currently working on a major revision (v2) of OML that includes:

Here is the current draft of the [OML2 proposal](https://www.opencaesar.io/blog/2023/02/01/OML-v2-Proposal.html).

Please leave your feedback below.

**Comment by melaasar** [comment]
The 2.0 release was made in May 2023 with the blog ([OML v2](https://www.opencaesar.io/projects/2023-7-21-OML-v2.html)) summarizing the changes from v1.

**Comment by raks0009** [comment]
Hi,
Is it possible to implement custom BuiltIn functions? 
If it is, I request you to also add an explanation for how to do so.
As I understand it right now, only the SWRLB builtin functions can be used.
Thanks again

**Comment by melaasar** [comment]
Correct. Currently, it's only possible to use the SWRL builtins.
I will look into whether the openllet reasoner allows custom builtins. If there is a hook, I will look into supporting it. I will comment with my finding soon.

**Comment by melaasar** [comment]
I confirmed that the [openllet](https://github.com/Galigator/openllet) reasoner currently only supports a subset of the standard SWRL builtins (I am revising the swrlb library in [core-vocabularies](https://github.com/opencaesar/core-vocabularies/blob/master/src/oml/www.w3.org/2003/11/swrlb.oml) to only retain those supported).  

Moreover, openllet currently does not support an API to extend the builtins out-of-the-box. However, there might be be a way to extend the reasoner programmatically to add such API, which could then be leveraged by openCAESAR. Feel free to open a user story on [owl-tools](https://github.com/opencaesar/owl-tools) for this.
